### PR TITLE
Added dispatch command

### PR DESCRIPTION
**Pull request by noirbizarre**
This command allows te dispatch/reaffect datasets or reuses to different user or organizations from a CSV file using the transfer API.

The command sniff the CSV and prompt the current user for the fields meaning.

### CODE REVIEW

File: ucli/commands/dispatch.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+
+TYPE_CHOICES, (IS_DATASET, IS_REUSE) = choice_enum('Dataset', 'Reuse')
+
+TARGET_CHOICES, (ORG_TARGET, USER_TARGET) = choice_enum('Organizations', 'Users')
+
+
+@cli.command()
+@click.option('--dryrun', '-d', is_flag=True)
+@click.option('--Force', '-f', is_flag=True)
```
Comment (abulte) [inline ucli/commands/dispatch.py]: Why upper case `F`?

File: ucli/commands/dispatch.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    item_col = reader.fieldnames[int(response)]
+    item_type = prompt_choices('Item type', *TYPE_CHOICES)
+
+    response = prompt_choices('Target column', *choices)
+    target_col = reader.fieldnames[int(response)]
+    target_type = prompt_choices('Target type', *TARGET_CHOICES)
+
+    message = click.prompt('Please enter the transfer reason')
+
+    if is_admin:
```
Comment (abulte) [inline ucli/commands/dispatch.py]: Maybe just `warning = ‘’` and then `if not admin:`

File: ucli/commands/dispatch.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        item_col=white(item_col),
+        target_type=white('organizations' if target_type == ORG_TARGET else 'users'),
+        target_col=white(target_col),
+        source=white(file.name),
+        message=message,
+        total=white(len(rows)),
+        warning=warning,
+    ))
+    if not force:
+        click.confirm('Are you sure ?', abort=True)
```
Comment (abulte) [inline ucli/commands/dispatch.py]: s/ ?/?/

File: ucli/commands/dispatch.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    # Perform
+    total = 0
+    item_endpoint = 'datasets/{id}/' if item_type == IS_DATASET else 'reuses/{id}/'
+    item_type_label = 'dataset' if item_type == IS_DATASET else 'reuse'
+    item_class = 'Dataset' if item_type == IS_DATASET else 'Reuse'
+    target_endpoint = 'organizations/{id}/' if target_type == ORG_TARGET else 'users/{id}/'
+    target_type_label = 'organization' if target_type == ORG_TARGET else 'user'
+    target_class = 'Organization' if target_type == ORG_TARGET else 'User'
+    target_fields = 'id,name' if target_type == ORG_TARGET else 'id,first_name,last_name'
+    for line, row in enumerate(rows, 2):  # Line 1 is header
```
Comment (abulte) [inline ucli/commands/dispatch.py:87]: TIL the second argument of `enumerate` 👍

**Review by abulte — APPROVED** [review]
Cool code, cool feature 👏

File: ucli/commands/dispatch.py (hunk 4 above)
Comment (noirbizarre) [reply ucli/commands/dispatch.py:87]: TIL `TIL`, so meta :joy:

Comment (abulte) [reply ucli/commands/dispatch.py:87]: 🤣

**Pull request merged**